posthog/api/exports.py

```diff
@@ -57,28 +61,74 @@ class ExportedAssetSerializer(serializers.ModelSerializer):
+            if not attrs.get("dashboard") and not attrs.get("insight"):
+                raise ValidationError("Either dashboard or insight is required for an export.")
 
-        if attrs.get("insight") and attrs["insight"].team.id != self.context["team_id"]:
-            raise ValidationError({"insight": ["This insight does not belong to your team."]})
+            if attrs.get("dashboard") and attrs["dashboard"].team.id != self.context["team_id"]:
+                raise ValidationError({"dashboard": ["This dashboard does not belong to your team."]})
+
+            if attrs.get("insight") and attrs["insight"].team.id != self.context["team_id"]:
+                raise ValidationError({"insight": ["This insight does not belong to your team."]})
 
         return attrs
 
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                "event",
+                OpenApiTypes.STR,
+                description="If exporting an event query to CSV. Filter events by event. For example `user sign up` or `$pageview`.",
+            ),
+            OpenApiParameter(
+                "person_id",
+                OpenApiTypes.INT,
+                description="If exporting an event query to CSV. Filter events by person id.",
+            ),
+            OpenApiParameter(
+                "distinct_id",
+                OpenApiTypes.INT,
+                description="If exporting an event query to CSV. Filter events by distinct id.",
+            ),
+            OpenApiParameter(
+                "before",
+                OpenApiTypes.DATETIME,
+                description="If exporting an event query to CSV. Only return events with a timestamp before this time.",
+            ),
+            OpenApiParameter(
+                "after",
+                OpenApiTypes.DATETIME,
+                description="If exporting an event query to CSV. Only return events with a timestamp after this time.",
+            ),
+            PropertiesSerializer(required=False),
+        ],
+    )
     def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> ExportedAsset:
         request = self.context["request"]
         validated_data["team_id"] = self.context["team_id"]
 
+        is_csv_export = validated_data.get("export_format") == ExportedAsset.ExportFormat.CSV
+        if is_csv_export:
+            validated_data["export_context"] = {
+                "file_export_type": "list_events",  # hard code to just this one for now
+                "filter": Filter(request=request, team=Team.objects.get(pk=validated_data["team_id"])).to_dict(),
+                "request_get_query_dict": request.GET.dict(),
+                "order_by": parse_order_by(request.GET.get("orderBy")),
+                "action_id": request.GET.get("action_id"),
+            }
         instance: ExportedAsset = super().create(validated_data)
 
-        task = insight_exporter.export_insight.delay(instance.id)
+        if is_csv_export:
```

Comment:
Thinking this _could_ be more nicely done in the exporter task itself. Have one task for "do the export" which decides based on the ExportedAsset which strategy to use. That way the caller doesn't need to be aware of the type of export?

Reply:
Yep, so eventually this would know how to to turn the request into some usable class/blob of possible things. Then the parent task takes the blob of things and figures out which actual export is being requested 👍